plans/bitswap-tuning/utils/net.go

```diff
@@ -0,0 +1,48 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/ipfs/testground/sdk/runtime"
+	"github.com/ipfs/testground/sdk/sync"
+)
+
+// SetupNetwork instructs the sidecar (if enabled) to setup the network for this
+// test case.
+func SetupNetwork(ctx context.Context, runenv *runtime.RunEnv, watcher *sync.Watcher, writer *sync.Writer) error {
+	if !runenv.TestSidecar {
+		return nil
+	}
+
+	// Wait for the network to be initialized.
+	if err := sync.WaitNetworkInitialized(ctx, runenv, watcher); err != nil {
+		return err
+	}
+
+	// TODO: just put the hostname inside the runenv?
```

Comment:
We can certainly do that with the local runners, but not sure how this would work with k8s. @nonsense?

Reply:
By the way I copied this from the DHT testplan, as with a bunch of the code in this utils package. It would be good to put it all in a central location somewhere

Reply:
This will be problematic in case we run concurrently the same testplan in multiple namespaces in the future. `pod` names are unique within a `namespace`, but we have `sidecar` running per host right now. I wouldn't worry too much about it, but maybe change the comment to say `unique testplan id`, rather than `hostname`, because as far as I understand we just need a unique testplan identifier here, so that we have a unique subtree, and we use `hostname` as such.

Reply:
Addressed in https://github.com/ipfs/testground/pull/353